src/__tests__/unknown.ts

```diff
@@ -65,28 +65,6 @@ export interface TestSchema {
 }`);
   });
 
-  test('`pattern(Joi.string(), Joi.number())`', () => {

```

Comment:
what happens if these removed tests are kept? this may cause regression for some people

Reply:
@mrjono1 
There is the following problems with supporting old tests:

```
 FAIL  src/__tests__/unknown.ts (11.936 s)
  ● test `Joi.unknown()` › `pattern(Joi.string(), Joi.number())`

    expect(received).toBe(expected) // Object.is equality

    - Expected  - 4
    + Received  + 1

      /**
       * a test schema definition
       */
      export interface TestSchema {
        name?: string;
    -   /**
    -    * Number Property
    -    */
    -   [x: string]: number;
    +   [pattern: string]: number;
      }

      76 |     const result = convertSchema({ sortPropertiesByName: false }, schema);
      77 |     expect(result).not.toBeUndefined;
    > 78 |     expect(result?.content).toBe(`/**
         |                             ^
      79 |  * a test schema definition
      80 |  */
      81 | export interface TestSchema {

      at Object.<anonymous> (src/__tests__/unknown.ts:78:29)

  ● test `Joi.unknown()` › `pattern(Joi.string(), Joi.AnySchema())`

    expect(received).toBe(expected) // Object.is equality

    - Expected  - 1
    + Received  + 1

      /**
       * a test schema definition
       */
      export interface TestSchema {
    -   [x: string]: {
    +   [pattern: string]: {
          id: string;
        }[];
      }

      123 |     const result = convertSchema({}, schema);
      124 |     expect(result).not.toBeUndefined;
    > 125 |     expect(result?.content).toBe(`/**
          |                             ^
      126 |  * a test schema definition
      127 |  */
      128 | export interface TestSchema {

      at Object.<anonymous> (src/__tests__/unknown.ts:125:29)
```
Pattern supporting changes this generation logic and we get a new output schema